src/main/java/seedu/address/model/person/OfficeHour.java

```diff
@@ -75,8 +83,31 @@ public boolean equals(Object other) {
                 || (other instanceof OfficeHour
                 && value.equals(((OfficeHour) other).value));
     }
+
     @Override
     public int hashCode() {
         return value.hashCode();
     }
+
+    /**
+     * Converts the office hour stored in value to what the user inputted.
+     * This is used for searching for office hours.
+     * @return The user input form of the office hour stored in value.
+     * @throws ParseException This exception will not be thrown as the value will always be in the correct format.
+     */
+    public String getUserInput() throws ParseException {
```

Comment:
This `parse exception` will never be thrown because `value` will always be in the right format as it goes through checks when users add/edit a contact's `office hours`.

Regardless, I am not sure if this good practice - if anyone has any suggestions or alternatives please let me know!

Reply:
Was the parse exception already there like from the start. Or it was implemented by one of us?

Reply:
No, it's not one of our exceptions. Java requires handling of `ParseException`s if the `parse` method of the `Date` class is called.

Gareth also uses it and handles it for the `office hour` class. 

In this case the exception will never be thrown because the input will already be validated.

Reply:
If java requres it we cant remove it either ways right?

Reply:
No we can't remove it, but I was just wondering if unhandled exceptions, although never thrown, is good practice.

Maybe there is a better way of implementing find by office hour.

Reply:
Following the advice of this [stackoverflow post](https://stackoverflow.com/a/30824035), I reckon that so long as we understand that the exception will not be thrown here, then there is no harm doing a `try-catch` where the `catch` simply gives an error like `this should not happen`. If an error does occur then we know our assumptions are violated elsewhere.

Reply:
Ah, that makes sense. I guess we'll go with that. Thanks @JJoeYi !